### ISSUE TITLE
predict to data frame rather than raster stack

### ISSUE THREAD

**Issue by AMBarbosa** [cjcarlson/embarcadero#13]
Thanks for the great package and vignette. I'm a bit puzzled that input data (including species occurrences and predictor variables) for `bart` are a data frame (`occ.df`), but then for `predict` you need a `RasterStack` of the predictor variables. Any chance of allowing 'climate' to be a data frame just like `occ.df`?

**Comment by AMBarbosa** [comment]
I did some quick edits to function [predict2.bart](https://github.com/cjcarlson/embarcadero/blob/master/R/predict.R), saved as function [predict_bart_df](https://github.com/AMBarbosa/unpackaged/blob/master/predict_bart_df) and it seems to be working on my species+variables data frame, at least with the default values. Please let me know if you happen to spot any problems, and feel free to use if interested.

**Comment by cjcarlson** [comment]
Howdy! If I'm understanding correctly - you could just use the native dbarts predict function, which is for prediction on a data frame. I wrote predict2.bart as a wrapper around that so I could predict on raster data - but I think what you want to do is just the base prediction function? Minus bells and whistles like a progress bar I guess

**Comment by AMBarbosa** [comment]
Thanks for your response, and sorry it took me a while to be able to get back to this. I had tried that, but if I use `predict` on a model made with embarcadero (or with the embarcadero package loaded?), it always fetches the `predict2.bart` version, and if the predictors are in a data.frame rather than a RasterStack, it fails with "Error in .subset2(x, i, exact = exact): recursive indexing failed at level 2". I also tried `dbarts::predict` and `predict.dbarts` (none of which exist), and then `stats::predict`, but instead of a vector of N predicted values (which would be equivalent to a raster layer with N pixels containing values), it returns a matrix with N columns and 1000 rows. I can obtain the expected vector if I then do `apply(predictions, 2, mean)`, though that's a bit convoluted (for R-newbie students especially). The only way I could get predictions directly as expected on an embarcadero model when predicting on a dataframe, was by using this separate [predict_bart_df](https://github.com/AMBarbosa/unpackaged/blob/master/predict_bart_df) function. I think it woud be fairly easy (and useful) to incorporate this dataframe functionality in `predict2.bart`, either by checking the class of `x.layers` and proceeding accordingly, or by adding an alternative `x.df` argument. But that's entirely up to you, of course! Just an idea :)  I'm getting very good SDM predictions with embarcadero and starting to include it in my training courses.
